module/solutions/helper.js

```diff
@@ -1807,6 +1829,89 @@ module.exports = class SolutionsHelper {
     });
   }
 
+  /**
+   * privateProgramAndSolutionDetails
+   * @method
+   * @name PrivateProgramAndSolutionDetails
+   * @param {Object} solutionData - solution data.
+   * @param {String} userId - user Id.
+   * @param {String} userToken - user token.
+   * @returns {Object} - Details of the private solution.
+   */
+
+  static privateProgramAndSolutionDetails(
+    solutionData,
+    userId = "",
+    userToken = ""
+  ) {
+    return new Promise(async (resolve, reject) => {
+      try {
+        let result = {};
+        // Check if a private program and private solution already exist or not for this user.
+        let privateSolutionDetails = await this.solutionDocuments(
+          {
+            parentSolutionId: solutionData.solutionId,
+            author: userId,
+            type: solutionData.type,
+            isAPrivateProgram: true,
+          },
+          ["_id","programId", "programName"]
+        );
+        if ( !privateSolutionDetails.length > 0 ) {
+          // Data for program and solution creation
+          let programAndSolutionData = {
+            type: constants.common.IMPROVEMENT_PROJECT,
+            subType: constants.common.IMPROVEMENT_PROJECT,
+            isReusable: false,
+            solutionId: solutionData.solutionId,
+          };
+  
+          if (solutionData.programName && solutionData.programName !== "") {
```

Comment:
@VISHNUDAS-tunerlabs If a solution has Id then it will have a name as the second check can be avoided